File: src/types/user/fi_user.ts
import { z } from 'zod';
import { timestampSchema } from '@/types/timestamp';
import { phoneNumberSchema } from '@/types/phone_number';
import { ObjectId } from 'mongodb';

export const userRoleSchema = z.enum(['sudo', 'admin', 'user']);

export const fiUserSchema = z.object({
  _id: z.custom<ObjectId>().optional(),
  phoneNumber: phoneNumberSchema,
  name: z.string().min(3, "En az 3 karakter gereklidir."),
  surname: z.string().min(2, "En az 2 karakter gereklidir."),
  companyId: z.string().optional(),
  role: userRoleSchema,
  createdAt: timestampSchema
});

export type FIUser = z.infer<typeof fiUserSchema>;
export type UserRole = z.infer<typeof userRoleSchema>;

File: fisoft/src/main/kotlin/types/FIUser.kt
package com.ilyasdemirkiran.types

import kotlinx.serialization.Serializable
import org.bson.codecs.pojo.annotations.BsonId

@Serializable
enum class UserRole {
    sudo, admin, user
}

@Serializable
data class FIUser(
    @BsonId val _id: String,
    val phoneNumber: String = "",
    val name: String = "",
    val surname: String = "",
    val companyId: String? = null,
    val role: UserRole = UserRole.user,
    val createdAt: String = ""
) {
    fun isAdmin(): Boolean = role == UserRole.admin || role == UserRole.sudo
}